pkg/operator/options/options.go

```diff
@@ -61,31 +62,31 @@ func (s *nodeIdentitiesValue) String() string { return strings.Join(*s, ",") }
 type optionsKey struct{}
 
 type Options struct {
-	ClusterName                    string
-	ClusterEndpoint                string // => APIServerName in bootstrap, except needs to be w/o https/port
-	VMMemoryOverheadPercent        float64
-	ClusterID                      string
-	KubeletClientTLSBootstrapToken string // => TLSBootstrapToken in bootstrap (may need to be per node/nodepool)
-	SSHPublicKey                   string // ssh.publicKeys.keyData => VM SSH public key // TODO: move to v1beta1.AKSNodeClass?
-
-	NetworkPlugin     string // => NetworkPlugin in bootstrap
-	NetworkPolicy     string // => NetworkPolicy in bootstrap
-	NetworkPluginMode string // => Network Plugin Mode is used to control the mode the network plugin should operate in. For example, "overlay" used with --network-plugin=azure will use an overlay network (non-VNET IPs) for pods in the cluster. Learn more about overlay networking here: https://learn.microsoft.com/en-us/azure/aks/azure-cni-overlay?tabs=kubectl#overview-of-overlay-networking
-	NetworkDataplane  string
-
-	NodeIdentities          []string // => Applied onto each VM
-	KubeletIdentityClientID string   // => Flows to bootstrap and used in drift
-	VnetGUID                string   // resource guid used by azure cni for identifying the right vnet
-	SubnetID                string   // => VnetSubnetID to use (for nodes in Azure CNI Overlay and Azure CNI + pod subnet; for for nodes and pods in Azure CNI), unless overridden via AKSNodeClass
+	ClusterName                    string  `json:"clusterName,omitempty"`
```

Comment:
The one risk I see here is that this requires users to opt-out of logging rather than opt-in.

If we want opt-in logging there are some dependencies that support it, although none of them are amazingly well adopted from what I can tell.